### PR TITLE
Adds CircularElement class to UI Hierarchy

### PR DESCRIPTION
**Pull request by cosparks**
- refactors some of the VisualElement width/height-related methods to RectangularElement class
- Circle positions refers to its top-left 'corner' -- (x, y) = (center - r, center - r)

### CODE REVIEW

File: src/ui/CircularElement.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    VisualElement(display) { }
+
+CircularElement::~CircularElement() { }
+
+int16_t CircularElement::getWidth() {
+    return _radius * 2;
+}
+
+int16_t CircularElement::getHeight() {
+    return _radius * 2;;
```
Comment (LaPommeCosmique) [inline src/ui/CircularElement.cpp]: theres an extra semicolon here;;

**Review by LaPommeCosmique — APPROVED** [review]

File: src/ui/CircularElement.cpp (hunk 1 above)
Comment (cosparks) [reply src/ui/CircularElement.cpp]: oh thanks good catch

compiler didn't seem to mind, I guess it probably ignored it

Comment (cosparks) [reply src/ui/CircularElement.cpp]: https://stackoverflow.com/questions/36408249/why-putting-2-semicolon-is-not-an-error

Comment (cosparks) [reply src/ui/CircularElement.cpp]: It makes sense I suppose.  I never realized it wouldn't even come up as a warning though

Comment (LaPommeCosmique) [reply src/ui/CircularElement.cpp]: lol, guess an empty statement is still a statement

**Pull request merged**